### DISCUSSION TITLE
Error: "User cancelled hostkey check" in ImageUpdateAutomation with Azure DevOps Repos

### DISCUSSION THREAD

**Discussion by funkymcb** [fluxcd/flux2#882] [Q&A] [answered]
Hello there, 
i tried to find a Solution to my Problem in the discussion but i am not quite sure if i found an answer. Maybe my problem will be solved after Issue 11 from the Roadmap?

The Problem is that i can't establish an ImageUpdateAutomation with a Repo on Azure DevOps.
When using a GitRespository Object with ssh authentication i get Events:
` unable to clone 'ssh://git@ssh.dev.azure.com/v3/path/to/repo', error : user cancelled hostkey check`
And with basic auth i get:
`[...] error: remote authentication required but no callback set`

Flux-System is on Azure DevOps Repos aswell using libgit2 implementation as described in the installation guide. 
Image registry (docker hub for now, later AzureCR) works fine with my Policies.

Thx for the help :)

**Comment by stefanprodan** [comment]
This was reported here also: https://github.com/fluxcd/image-automation-controller/issues/71#issuecomment-773432

**Reply by funkymcb** [reply]
Thx a lot

**Comment by stefanprodan** [comment, marked as answer]
Fixed in image-automation-controller v0.5.0 https://github.com/fluxcd/image-automation-controller/blob/main/CHANGELOG.md#050

**Comment by maxgio92** [comment]
Hi all, I'm experiencing the same issue with authentication to `GitRepository` with HTTP basic authentication and `libgit2` (Azure DevOps Repos).
I didn't find related issues open, I'd like to debug the issue before opening a new one.
In the meantime, do you have some info regarding this case?
Thank you all!

**Reply by pjbgf** [reply]
@maxgio92 would you mind confirming whether this is still the case for you? I have recently tested this using version `ghcr.io/fluxcd/image-automation-controller:v0.21.1` and I could not reproduce the issue.

**Reply by maxgio92** [reply]
Hi @pjbgf, thanks for your time. Unfortunately now I'm not able to neither reproduce nor test that version with same scenario
